File: LeetCode/476. Number Complement.py
### KNOWLEDGE IS POWER  ###

class Solution(object):
    def findComplement(self, num):
        """
        :type num: int
        :rtype: int
        """
        mask, tmp = 0, num
        while tmp:
            mask = (mask << 1) | 1
            tmp >>= 1
        return mask ^ num

File: LeetCode/476. Number Complement.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    int findComplement(int num) {
        int mask = 0;
        for(int tmp = num; tmp; tmp >>= 1)
            mask = (mask << 1) | 1;      
        return mask ^ num;
    }
};
